### PR TITLE
[CWG 1] P2796R0 except issues 2518, 2521, 2659, 2674, 2678, and 2691 

### LINKED ISSUE #6083: [2023-02 CWG Motion 1] P2796R0 CWG ready issues except 2518, 2521, 2659, 2674, 2678, and 2691

### LINKED ISSUE #6112: [expr.const] Missing indefinite article in CWG2558 resolution
The resolution of CWG2558 adds:

> if the value is an object of scalar type, it does not have indeterminate value (6.7.4 [basic.indet]),

but this should be:

> if the value is an object of scalar type, it does not have <ins>an</ins> indeterminate value (6.7.4 [basic.indet]),

because in C++ (unlike in C) an indeterminate value is a particular kind of value, rather than "has indeterminate value" being a property of an object / region of storage. See [basic.indet]p1.

### PR DESCRIPTION
**Pull request by burblebee**
Fixes #6083.
Partially addresses cplusplus/papers#1480
Fixes cplusplus/nbballot#479
Fixes #6112

### CODE REVIEW

File: source/expressions.tex [hunk 1]
```diff
@@ -6555,6 +6539,11 @@
 \tcode{q<=p}, and \tcode{q<p} all yield \keyword{true} and \tcode{p<=q},
 \tcode{p<q}, \tcode{q>=p}, and \tcode{q>p} all yield \keyword{false}.
 Otherwise, the result of each of the operators is unspecified.
+\begin{note}
+A relational operator applied
+to unequal function pointers or to unequal pointers to void
```
Comment (JohelEGP) [inline source/expressions.tex:6544]:
```suggestion
to unequal function pointers or to unequal pointers to \tcode{void}
```

File: source/expressions.tex [hunk 2]
```diff
@@ -7785,6 +7782,11 @@
   if the value is an object of class type,
   each non-static data member of reference type refers to
   an entity that is a permitted result of a constant expression,
+
+  \item
+  if the value is an object of scalar type,
+  it does not have indeterminate value\iref{basic.indet},
```
Comment (JohelEGP) [inline source/expressions.tex:7788]: #6112 suggests
```suggestion
  it does not have an indeterminate value\iref{basic.indet},
```

File: source/classes.tex [hunk 3]
```diff
@@ -2095,7 +2095,7 @@
 
 \pnum
 A defaulted destructor is a constexpr destructor
-if it satisfies the requirements for a constexpr function\iref{dcl.constexpr}.
+if it is constexpr-suitable\iref{dcl.constexpr}.
```
Comment (JohelEGP) [inline source/classes.tex:2098]: I don't see this change from CWG2602:
> Change in 9.5.2 [[dcl.fct.def.default](https://wg21.link/dcl.fct.def.default#3)] paragraph 3 as follows:
>
>    A function explicitly defaulted on its first declaration is implicitly inline (9.2.8 [[dcl.inline](https://wg21.link/dcl.inline)]), and is implicitly constexpr (9.2.6 [[dcl.constexpr](https://wg21.link/dcl.constexpr)]) if it satisfies the requirements for a constexpr function is constexpr-suitable.

**Review by JohelEGP — CHANGES_REQUESTED** [review]

File: source/expressions.tex (hunk 1 above)
Comment (burblebee) [reply source/expressions.tex:6544]: done, thanks!

**Review by JohelEGP — APPROVED** [review]
Also resolves #6112.

**Pull request merged**